testsuite/shared/src/main/java/org/wildfly/test/stability/StabilityServerSetupTasks.java

```diff
@@ -0,0 +1,133 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wildfly.test.stability;
+
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.integration.management.ManagementOperations;
+import org.jboss.as.test.integration.management.util.ServerReload;
+import org.jboss.as.version.Stability;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.wildfly.core.testrunner.ManagementClient;
+import org.wildfly.core.testrunner.ServerSetupTask;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.CORE_SERVICE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.STABILITY;
+import static org.jboss.as.server.controller.descriptions.ServerDescriptionConstants.SERVER_ENVIRONMENT;
+
+/**
+ * For tests that need to run under a specific server stability level,
+ * the server setup tasks from the inner classes can be used to change the stability level of the server to the desired level.
+ * Once the test is done, the original stability level is restored.
+ */
+public abstract class StabilityServerSetupTasks implements ServerSetupTask {
+
+    private final Stability desiredStability;
+    private volatile Stability originalStability;
+
+
+    public StabilityServerSetupTasks(Stability desiredStability) {
+        this.desiredStability = desiredStability;
+    }
+
+    @Override
+    public void setup(ManagementClient managementClient) throws Exception {
+        // We can't reload to default since that will not have the stability paremeter in the reload operation
+        Assume.assumeFalse(desiredStability.equals(Stability.DEFAULT));
+
+        reloadToDesiredStability(managementClient.getControllerClient(), desiredStability);
+    }
+
+    @Override
+    public void tearDown(ManagementClient managementClient) throws Exception {
+        reloadToDesiredStability(managementClient.getControllerClient(), originalStability);
+    }
+
+    private Stability reloadToDesiredStability(ModelControllerClient client, Stability stability) throws Exception {
+        // Check the stability
+        Stability currentStability = readCurrentStability(client);
+        if (originalStability == null) {
+            originalStability = currentStability;
+        }
+
+        // The stability parameter for the reload opration is only registered below the default level
+        Assume.assumeFalse("Can't reload to a different stability when server stability level is default", currentStability == Stability.DEFAULT && stability != Stability.DEFAULT);
```

Comment:
This should do a read-operation-names and look for the op and Assume.assumeTrue(opnames.contains(theopname)).

IOW don't hard code the current rules about the availability of the op into this test fixture.

Reply:
We also need https://issues.redhat.com/browse/WFCORE-6731 and then this can do an Assume.assumeTrue(supported.contains(stability))

Reply:
@bstansberry I'm not totally sure about WFCORE_6731 for this? I am able to start WildFly Core with `--stability=default` so I presume that means default is a legal stability level. Still, I can add a placeholder for this check, since there may well be something which isn't turned on in my version of WildFly yet enforcing this. 

The issue I am trying to solve here, is that if we were to reload to default to run the test, we would not be able to reload back to community since the new reload operation is only registered at that level. 

I think I will check the stability of the reload-enhanced operation definition, and check that it would still be available if the reload to the intended stability takes place.

Reply:
@kabir OK. I think this is fine for WF 32, but please file a JIRA to improve it so we don't forget.